eclipse/src/saros/ui/command_handlers/SessionAddSelectedProjectsHandler.java

```diff
@@ -18,7 +27,8 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
     List<IResource> selectedResources =
         SelectionRetrieverFactory.getSelectionRetriever(IResource.class).getSelection();
 
-    CollaborationUtils.addResourcesToSession(selectedResources);
+    WizardUtils.openAddResourcesToSessionWizard(new HashSet<>(selectedResources));
```

Comment:
This is very inconvenient now. The correct rework would be to add the possibility to simply right-click on a folder or project and the click share.

Reply:
Yes, the usability is worse, but not by a lot in my opinion. Instead of directly adding the resource to the session, it now opens a dialog where the chosen resources are pre-selected. It is one additional click.

The issue with keeping the old logic is that the user can select arbitrary sets of resources, parts of which could already be shared or could be children of other selected resources. To process this, it would require applying all of the filtering logic done in the wizard. Furthermore, due to the filtering, the actually shared resources might not match the users expectations. But, as all of this happens in the background, we would not have any way of displaying this to the user. With the wizard, the user sees the actual result that the addition of the resources would have/which resources would actually be added to the session.